### PR TITLE
Model CapabilityAuthServicePredicate

### PR DESCRIPTION
**Pull request by hchoura**

### CODE REVIEW

**Review by Nauna — APPROVED** [review]

File: src/foam/mlang/predicate/CapabilityAuthServicePredicate.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      name: 'f',
+      args: [
+        {
+          name: 'obj',
+          type: 'Object'
+        }
+      ],
+      javaCode: `
+          foam.core.X x = getX();
+          Logger logger = (Logger) getX().get("logger");
```
Comment (kgrgreer) [inline src/foam/mlang/predicate/CapabilityAuthServicePredicate.js]: You could be a little more efficient by only looking up the logger if/when you need it before line 57.

File: src/foam/mlang/predicate/CapabilityAuthServicePredicate.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'CapabilityAuthServicePredicate',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  documentation: `
+    Create modeled predicate.
```
Comment (kgrgreer) [inline src/foam/mlang/predicate/CapabilityAuthServicePredicate.js]: All models are modelled, so maybe a better description could go here.

File: src/foam/mlang/predicate/CapabilityAuthServicePredicate.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          name: 'obj',
+          type: 'Object'
+        }
+      ],
+      javaCode: `
+          foam.core.X x = getX();
+          UserCapabilityJunction ucj = (UserCapabilityJunction) obj;
+          if ( ucj.getStatus() == CapabilityJunctionStatus.GRANTED ) {
+            Capability c = (Capability) getCapabilityDAO().find(ucj.getTargetId());
+            if ( c != null &&
```
Comment (kgrgreer) [inline src/foam/mlang/predicate/CapabilityAuthServicePredicate.js]: This indentation is weird and hard to read, just put the whole expression on one line as it fits easily.

File: src/foam/mlang/predicate/CapabilityAuthServicePredicate.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'CapabilityAuthServicePredicate',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  documentation: `
+    The main reason to  create this modeled predicate is to solve a performance
+    issue. We need to avoid using anonymous inner class.
+    By creating a model, we will be able to have a better concurrently performance.
```
Comment (jlhughes) [inline src/foam/mlang/predicate/CapabilityAuthServicePredicate.js]:
```suggestion
    By creating a model, we will be able to have a better concurrency performance.
```

**Pull request merged**